test-data/unit/check-final.test

```diff
@@ -1250,3 +1250,28 @@ def check_final_init() -> None:
     new_instance = FinalInit()
     new_instance.__init__()
 [builtins fixtures/tuple.pyi]
+
+[case testNarrowingOfFinalPersistsInFunctions]
+from typing import Final, Union
+
+def _init() -> Union[int, None]:
+    return 0
+
+FOO: Final = _init()
+# For some reason the following isn't narrowed at all
+# (always stays int | None):
+# FOO: Final[Union[int, None]] = 0
```

Comment:
```suggestion
```
This is probably just a variant of https://github.com/python/mypy/issues/2008

Reply:
It is, thanks!